pyaedt/modules/solutions.py

```diff
@@ -1071,338 +1075,253 @@ def ifft_to_file(
-        rows = []
-        cols = []
-        for portstring in self.all_port_names:
-            index_str = self.get_array_index(portstring)
-            rows.append(index_str[1])
-            cols.append(index_str[0])
-
-        row_min = np.min(rows)
-        row_max = np.max(rows)
-        col_min = np.min(cols)
-        col_max = np.max(cols)
-        return [col_min, col_max, row_min, row_max]
-
-    @pyaedt_function_handler()
-    def array_center_and_edge(self):
-        """
-        Find the center and edge of our array, assuming all ports in far field
-        mapping file are active ports.
-
-        Returns
-        -------
-        bool
-        """
-        AMax = 0
-        BMax = 0
-        RMax = 0
-        XMax = 0
-        YMax = 0
-        CenterA = 0
-        CenterB = 0
-        CenterX = 0
-        CenterY = 0
-
-        # collecting all active cells inside the specified region
-        activeCells = []
-
-        for i in range(0, len(self.all_port_names)):
-            index_str = self.get_array_index(self.all_port_names[i])
-            row = index_str[1]
-            col = index_str[0]
-            a = row
-            b = col
-
-            activeCells.append((a, b))  # because ffd is assuming all ffd files are active
-        if len(activeCells) == 0:
-            return
-
-        [a_min, a_max, b_min, b_max] = self.array_min_max_values()
-
-        CenterA = (a_min + a_max) / 2
-        CenterB = (b_min + b_max) / 2
-        CenterX = (CenterA + 0.5) * self.Ax + (CenterB + 0.5) * self.Bx
-        CenterY = (CenterA + 0.5) * self.Ay + (CenterB + 0.5) * self.By
-
-        self.CenterA = CenterA
-        self.CenterB = CenterB
-        self.CenterX = CenterX
-        self.CenterY = CenterY
-        # find the distance from the edge to the center
-        AMax = a_max - a_min
-        BMax = b_max - b_min
-
-        self.AMax = AMax
-        self.BMax = BMax
-        for a, b in activeCells:
-            x = (a + 0.5) * self.Ax + (b + 0.5) * self.Bx
-            y = (a + 0.5) * self.Ay + (b + 0.5) * self.By
-            x_dis = abs(x - CenterX)
-            y_dis = abs(y - CenterY)
-            distance = math.sqrt(x_dis**2 + y_dis**2)
-            XMax = max(XMax, x_dis)
-            YMax = max(YMax, y_dis)
-            RMax = max(RMax, distance)
-
-        self.RMax = RMax
-        self.XMax = XMax
-        self.YMax = YMax
-        self.RMax *= 2
-        self.XMax *= 2
-        self.YMax *= 2
-        return True
+            self.farfield_data = self.combine_farfield()
 
-    @pyaedt_function_handler()
-    def element_location(self, a, b):
-        """Element location in the array.
-
-        Parameters
-        ----------
-        a : int
-        b : int
-
-        Returns
-        -------
-        list of float
+    @property
+    def taper(self):
+        """Taper type.
+
+        Options are:
+        - "flat"
+        - "uniform"
+        - "cosine"
+        - "triangular"
+        - "hamming"
```

Comment:
```suggestion
        - ``"cosine"``
        - ``"flat"``
        - ``"hamming"``
        - ``"triangular"``
        - ``"uniform"``
```
Options should be tagged as code entities and listed alphabetically for easier scanning.